### PR TITLE
[API] Save literals to a file while merging the snapshots

### PR DESCRIPTION
**Pull request by TamasZakor**
Add command line options to snapshot merge:
--save-literals-list-format
--save-literals-c-format

JerryScript-DCO-1.0-Signed-off-by: Tamas Zakor ztamas@inf.u-szeged.hu

### CODE REVIEW

**Comment by zherczeg** [comment]
This patch looks like an API change. Is this flagged as one?

File: docs/02.API-REFERENCE.md [hunk 1]
```diff
@@ -5317,15 +5317,23 @@ jerry_parse_and_save_literals (const jerry_char_t *source_p,
                                bool is_strict,
                                uint32_t *buffer_p,
                                size_t buffer_size,
-                               bool is_c_format);
+                               bool is_c_format,
+                               const char *save_literals_file_name_p,
```
Comment (zherczeg) [inline docs/02.API-REFERENCE.md]: JerryScript does not know about the concept of files.

File: jerry-core/api/jerry-snapshot.c [hunk 2]
```diff
@@ -1198,157 +1198,6 @@ update_literal_offsets (uint8_t *buffer_p, /**< snapshot buffer start */
...
-#endif /* JERRY_ENABLE_SNAPSHOT_SAVE */
-
-/**
- * Merge multiple snapshots into a single buffer
- *
- * @return length of merged snapshot file
- *         0 on error
- */
-size_t
-jerry_merge_snapshots (const uint32_t **inp_buffers_p, /**< array of (pointers to start of) input buffers */
```
Comment (zherczeg) [inline jerry-core/api/jerry-snapshot.c]: Why this function is reordered?

Comment (TamasZakor) [reply jerry-core/api/jerry-snapshot.c]: Because some static functions is used in jerry_merge_snapshots.

File: docs/02.API-REFERENCE.md (hunk 1 above)
Comment (TamasZakor) [reply docs/02.API-REFERENCE.md]: What do you mean by this?

Comment (zherczeg) [reply docs/02.API-REFERENCE.md]: You cannot use fopen/fclose/etc. functions in JerryScript. Theis provides the most compatibility with low-end systems.

File: jerry-core/api/jerry-snapshot.c (hunk 2 above)
Comment (zherczeg) [reply jerry-core/api/jerry-snapshot.c]: I suspect the logic of merging and snapshot saving is combined in this patch. It should not.

**Pull request closed without merging**